blog/2015/09/thinking-about-teaching.html

```diff
@@ -0,0 +1,360 @@
+---
+layout: blog
+root: ../../..
+author: Greg Wilson
+title: "Thinking About Teaching"
+date: 2015-09-25
+time: "02:00:00"
+category: ["Teaching"]
+---
+<!-- start excerpt -->
+<p>
+  A little over a year ago,
+  we blogged about <a href="{{page.root}}/blog/2014/09/building-better-teachers.html">jugyokenkyu</a>,
+  or "lesson study",
+  a bucket of practices that Japanese teachers use to hone their craft,
+  from observing each other at work
+  to discussing the lesson afterward
+  to studying curriculum materials with colleagues.
+  Getting the Software Carpentry Foundation up and running almost immediately pushed that aside,
+  but now that the SCF is up and running,
+  it's time to return to the subject.
+  Discussion of <a href="how-teaching-knowledge-is-transferred.html">how teaching practices are transferred</a>
+  is part of that;
+  so are two other developments this week.
+</p>
+<!-- end excerpt -->
+<p>
+  First, Victor Eijkhout has been thinking about
+  <a href="https://bitbucket.org/VictorEijkhout/parallel-computing-book/downloads/eijkhout-teachmpi.pdf">how to teach MPI</a>.
+  In particular,
+  he has been thinking about how the order in which concepts are usually introduced can mislead or confuse,
+  and about a different order that leads more naturally to proficient use.
+  We're not going to include MPI in Software Carpentry any time soon (if ever),
+  but I hope that the next major revision of our lessons will include this kind of design rationale.
+</p>
+<p>
+  Second,
+  <a href="{{page.root}}/pages/team.html#bostroem.a">Azalee Bostroem</a> surveyed people who have taught our current Python lesson
+  to find out what they're actually doing and how it has been going.
+  <a href="python-lesson-survey-results.pdf">The results</a> are fascinating&mdash;here
+  are some of the key findings, and my comments:
+</p>
+<dl>
+  <dt>
+    60% of our lessons are being taught to novices.
+  </dt>
+  <dd>
+    <p>
+      The term "novice" is a bit ambiguous here:
+      it can mean either "someone who has never programmed before"
```

Comment:
In the survey I explicitly defined novice as never having programmed in any language and intermediate as someone who has programmed in a language other than python, and advanced as someone who has programmed in python